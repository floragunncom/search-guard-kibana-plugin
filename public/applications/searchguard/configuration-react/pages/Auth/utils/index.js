/* eslint-disable @kbn/eslint/require-license-header */
export { default as resourcesToUiResources } from './resources_to_ui_resources';
export { getSideNavItems } from './get_side_nav_items';
