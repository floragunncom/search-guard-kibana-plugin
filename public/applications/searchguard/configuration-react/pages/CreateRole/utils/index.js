/*
 *    Copyright 2020 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export {
  roleToFormik,
  indexPermissionToUiIndexPermission,
  aliasPermissionToUiAliasPermission,
  tenantPermissionToUiTenantPermission,
  indicesToUiIndices,
  actionGroupsToUiClusterIndexTenantActionGroups,
  tenantsToUiTenants,
  excludeIndexPermissionToUiExcludeIndexPermission,
  maskedFieldsToUiMaskedFields,
} from './role_to_formik';
export { formikToRole } from './formik_to_role';
export { useIndexPatterns, indexPatternNames, renderIndexOption } from './use_index_patterns';
export { fieldNamesToUiFieldNames } from './field_names_to_ui_field_names';
export { mappingsToFieldNames } from './mappings_to_field_names';
export { validateMaskedFields } from './validate_masked_fields';
export { validateMaskedFieldsRegexMask } from './validate_masked_fields_regex_mask';
