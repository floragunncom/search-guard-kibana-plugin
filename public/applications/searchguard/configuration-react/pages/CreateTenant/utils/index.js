export { default as tenantToFormik } from './tenant_to_formik';
export { default as formikToTenant } from './formik_to_tenant';
