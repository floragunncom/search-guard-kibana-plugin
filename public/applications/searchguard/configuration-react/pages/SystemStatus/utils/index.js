export { default as getResource } from './get_resource';
