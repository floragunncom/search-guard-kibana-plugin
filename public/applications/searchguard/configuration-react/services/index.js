/* eslint-disable @kbn/eslint/require-license-header */
export { default as LocalStorageService } from './LocalStorageService';
export { default as ElasticsearchService } from './ElasticsearchService';

export { ApiService } from './ApiService';
export { InternalUsersService } from './InternalUsersService';
export { ActionGroupsService } from './ActionGroupsService';
export { TenantsService } from './TenantsService';
export { SgConfigService } from './SgConfigService';
export { RolesService } from './RolesService';
export { RolesMappingService } from './RolesMappingService';
export { ConfigService } from './ConfigService';
