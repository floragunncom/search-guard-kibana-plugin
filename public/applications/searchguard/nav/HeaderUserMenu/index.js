/* eslint-disable @kbn/eslint/require-license-header */
export { HeaderUserMenuApp } from './HeaderUserMenuApp';
