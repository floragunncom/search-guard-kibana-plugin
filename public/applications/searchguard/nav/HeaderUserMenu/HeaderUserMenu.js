/*
 *    Copyright 2021 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  EuiButtonEmpty,
  EuiToolTip,
  EuiPopover,
  EuiAvatar,
  EuiFlexGroup,
  EuiFlexItem,
  EuiText,
  EuiLink,
  EuiSpacer,
  EuiListGroup,
  EuiListGroupItem,
} from '@elastic/eui';
import { AccessControlService } from '../../../../services';
import { logoutText } from '../utils/i18n';
import {
  SEARCHGUARD_MULTITENANCY_APP_ID,
  SEARCHGUARD_MULTITENANCY_APP_TITLE,
} from '../../../multitenancy/utils/constants';
import {
  SEARCHGUARD_ACCOUNTINFO_APP_ID,
  SEARCHGUARD_ACCOUNTINFO_APP_TITLE,
} from '../../../accountinfo/utils/constants';

export function LogoutBtn({ onClick, authType }) {
  if (authType === 'kerberos' || authType === 'proxy') return null;
  return (
    <EuiLink onClick={onClick} aria-label="logout" data-test-subj="sg.userMenu.button-logout">
      {logoutText}
    </EuiLink>
  );
}

export function HeaderUserMenu({ httpClient, configService, kibanaAppService }) {
  const userNameText = configService.get('restapiinfo.user_name');
  const { type: authType, logout_url: logoutUrl } = configService.get('searchguard.auth', {});

  const [isOpen, setIsOpen] = useState(false);
  const acService = new AccessControlService({ httpClient, authType });

  const appMenu = [
    {
      label: SEARCHGUARD_ACCOUNTINFO_APP_TITLE,
      appId: SEARCHGUARD_ACCOUNTINFO_APP_ID,
      isEnabled: configService.get('searchguard.accountinfo.enabled', false),
    },
    {
      label: SEARCHGUARD_MULTITENANCY_APP_TITLE,
      appId: SEARCHGUARD_MULTITENANCY_APP_ID,
      isEnabled: configService.get('searchguard.multitenancy.enabled', false),
    },
  ];

  function openPopover() {
    setIsOpen((prevState) => !prevState);
  }

  function closePopover() {
    setIsOpen(false);
  }

  function logOut() {
    acService.logout({ logoutUrl });
  }

  function renderAppMenu(appMenu) {
    return appMenu
      .filter((item) => item.isEnabled)
      .map((item) => {
        return (
          <EuiListGroupItem
            data-test-subj={`sg.userMenu.appMenu.item.${item.appId}`}
            key={item.appId}
            label={item.label}
            onClick={() => {
              kibanaAppService.navigateToApp(item.appId);
            }}
          />
        );
      });
  }

  const button = (
    <EuiButtonEmpty onClick={openPopover} data-test-subj="sg.userMenu.button">
      <EuiAvatar size="s" name={userNameText} />
    </EuiButtonEmpty>
  );

  return (
    <EuiPopover
      style={{ paddingTop: 2 }}
      data-test-subj="sg.userMenu"
      repositionOnScroll
      isOpen={isOpen}
      button={button}
      closePopover={closePopover}
    >
      <div id="sg.userMenu.content" style={{ width: '400px' }}>
        <EuiFlexGroup>
          <EuiFlexItem grow={false}>
            <EuiAvatar name={userNameText} />
          </EuiFlexItem>
          <EuiFlexItem wrap={true}>
            <EuiToolTip position="bottom" content={userNameText}>
              <EuiText>
                <div style={{ wordBreak: 'break-word', overflowWrap: 'anywhere' }}>
                  {userNameText}
                </div>
              </EuiText>
            </EuiToolTip>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <LogoutBtn onClick={logOut} authType={authType} />
          </EuiFlexItem>
        </EuiFlexGroup>

        <EuiSpacer />
        <EuiListGroup flush={true} bordered={true}>
          {renderAppMenu(appMenu)}
        </EuiListGroup>
      </div>
    </EuiPopover>
  );
}

HeaderUserMenu.propTypes = {
  httpClient: PropTypes.object.isRequired,
  configService: PropTypes.object.isRequired,
  kibanaAppService: PropTypes.object.isRequired,
};
