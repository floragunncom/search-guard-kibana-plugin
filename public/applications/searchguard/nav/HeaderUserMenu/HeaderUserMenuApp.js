/*
 *    Copyright 2020 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { HeaderUserMenu } from './HeaderUserMenu';

export class HeaderUserMenuApp {
  start({ core, httpClient, configService } = {}) {
    const userName = configService.get('restapiinfo.user_name');
    const authType = configService.get('searchguard.auth.type');
    const logoutUrl = configService.get('searchguard.auth.logout_url');

    const props = { httpClient, authType };

    if (logoutUrl) {
      props.logoutUrl = logoutUrl;
    }

    if (userName) {
      props.userName = userName;
      props.userNameTooltipText = userName;
    }

    core.chrome.navControls.registerRight({
      order: 5000,
      mount: (element) => {
        ReactDOM.render(<HeaderUserMenu {...props} core={core} />, element);
        return () => ReactDOM.unmountComponentAtNode(element);
      },
    });
  }
}
