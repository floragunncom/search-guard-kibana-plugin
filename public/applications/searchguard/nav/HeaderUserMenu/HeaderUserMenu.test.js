/*
 *    Copyright 2021 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { HeaderUserMenu, LogoutBtn } from './HeaderUserMenu';
import { ConfigService } from '../../../../services';
import { setupKibanaCoreAppServiceMock } from '../../../../utils/mocks';

describe(LogoutBtn.name, () => {
  let renderer;

  beforeEach(() => {
    renderer = new ShallowRenderer();
  });

  test('render the button', () => {
    const tree = renderer.render(<LogoutBtn authType="any" onClick={() => null} />);
    expect(tree).toMatchSnapshot();
  });

  test('do not render the button if Kerberos auth', () => {
    const tree = renderer.render(<LogoutBtn authType="kerberos" onClick={() => null} />);
    expect(tree).toMatchSnapshot();
  });

  test('do not render the button if Proxy auth', () => {
    const tree = renderer.render(<LogoutBtn authType="proxy" onClick={() => null} />);
    expect(tree).toMatchSnapshot();
  });
});

describe(HeaderUserMenu.name, () => {
  let renderer;
  let httpClient;
  let kibanaAppService;

  beforeEach(() => {
    renderer = new ShallowRenderer();
    httpClient = {};
    kibanaAppService = setupKibanaCoreAppServiceMock();
  });

  test('can render the user menu', () => {
    const configService = new ConfigService({
      config: {
        restapiinfo: { user_name: 'a user' },
        searchguard: {
          auth: {
            type: 'basicauth',
            logout_url: '/logout',
          },
          accountinfo: { enabled: true },
          multitenancy: { enabled: true },
        },
      },
    });

    const tree = renderer.render(
      <HeaderUserMenu
        httpClient={httpClient}
        kibanaAppService={kibanaAppService}
        configService={configService}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  test('filter menu items (Multitenancy is disabled)', () => {
    const configService = new ConfigService({
      config: {
        restapiinfo: { user_name: 'a user' },
        searchguard: {
          auth: {
            type: 'basicauth',
            logout_url: '/logout',
          },
          accountinfo: { enabled: true },
          multitenancy: { enabled: false },
        },
      },
    });

    const tree = renderer.render(
      <HeaderUserMenu
        httpClient={httpClient}
        kibanaAppService={kibanaAppService}
        configService={configService}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
