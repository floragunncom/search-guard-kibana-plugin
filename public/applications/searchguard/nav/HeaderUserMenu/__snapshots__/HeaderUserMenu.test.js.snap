// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderUserMenu can render the user menu 1`] = `
<EuiPopover
  anchorPosition="downCenter"
  button={
    <EuiButtonEmpty
      data-test-subj="sg.userMenu.button"
      onClick={[Function]}
    >
      <EuiAvatar
        name="a user"
        size="s"
      />
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  data-test-subj="sg.userMenu"
  display="inlineBlock"
  hasArrow={true}
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="m"
  repositionOnScroll={true}
  style={
    Object {
      "paddingTop": 2,
    }
  }
>
  <div
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <EuiFlexGroup>
      <EuiFlexItem
        grow={false}
      >
        <EuiAvatar
          name="a user"
        />
      </EuiFlexItem>
      <EuiFlexItem
        wrap={true}
      >
        <EuiToolTip
          content="a user"
          delay="regular"
          position="bottom"
        >
          <EuiText>
            <div
              style={
                Object {
                  "overflowWrap": "anywhere",
                  "wordBreak": "break-word",
                }
              }
            >
              a user
            </div>
          </EuiText>
        </EuiToolTip>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <LogoutBtn
          authType="basicauth"
          onClick={[Function]}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
    <EuiSpacer />
    <EuiListGroup
      bordered={true}
      flush={true}
    >
      <EuiListGroupItem
        label="Account Info"
        onClick={[Function]}
      />
      <EuiListGroupItem
        label="Multitenancy"
        onClick={[Function]}
      />
    </EuiListGroup>
  </div>
</EuiPopover>
`;

exports[`HeaderUserMenu filter menu items (Multitenancy is disabled) 1`] = `
<EuiPopover
  anchorPosition="downCenter"
  button={
    <EuiButtonEmpty
      data-test-subj="sg.userMenu.button"
      onClick={[Function]}
    >
      <EuiAvatar
        name="a user"
        size="s"
      />
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  data-test-subj="sg.userMenu"
  display="inlineBlock"
  hasArrow={true}
  isOpen={false}
  ownFocus={false}
  panelPaddingSize="m"
  repositionOnScroll={true}
  style={
    Object {
      "paddingTop": 2,
    }
  }
>
  <div
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <EuiFlexGroup>
      <EuiFlexItem
        grow={false}
      >
        <EuiAvatar
          name="a user"
        />
      </EuiFlexItem>
      <EuiFlexItem
        wrap={true}
      >
        <EuiToolTip
          content="a user"
          delay="regular"
          position="bottom"
        >
          <EuiText>
            <div
              style={
                Object {
                  "overflowWrap": "anywhere",
                  "wordBreak": "break-word",
                }
              }
            >
              a user
            </div>
          </EuiText>
        </EuiToolTip>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <LogoutBtn
          authType="basicauth"
          onClick={[Function]}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
    <EuiSpacer />
    <EuiListGroup
      bordered={true}
      flush={true}
    >
      <EuiListGroupItem
        label="Account Info"
        onClick={[Function]}
      />
    </EuiListGroup>
  </div>
</EuiPopover>
`;

exports[`LogoutBtn do not render the button if Kerberos auth 1`] = `null`;

exports[`LogoutBtn do not render the button if Proxy auth 1`] = `null`;

exports[`LogoutBtn render the button 1`] = `
<EuiLink
  aria-label="logout"
  data-test-subj="sg.userMenu.button-logout"
  onClick={[Function]}
>
  <EuiI18n
    default="Logout"
    token="sg.nav.logout.logoutText"
  />
</EuiLink>
`;
